### DISCUSSION TITLE
Problem with connection via Com port

### DISCUSSION THREAD

**Discussion by Artisan401** [mayrthom/CartridgeWriter_uPrint#7] [Q&A] [answered]
Hello. 
I have a problem communicating with the printer. He doesn't want to communicate with me through the diagnostic port. Can't the printer have any protection or security?
We have this type of printer at school Stratasys uPrint SE Plus. 
We usually print on it, but the material is running out and we would like to use a cheaper type of ABS filament.
Thank you for any advice

**Comment by mayrthom** [comment, marked as answer]
Which error message does the program show?
Make sure, that you use a null model cable. It’s a common issue, that people use a normal serial cable instead of a null modem cable. 
If you are not sure which cable you are using, you can check with a multimeter.
Here you can find some further information:
https://en.wikipedia.org/wiki/Null_modem

Best Regards

Thomas

**Reply by Artisan401** [reply]
Thanks, I'll do screens with bugs tomorrow and send it. I'll find out what I have about the cable.

**Reply by Artisan401** [reply]
I got this cable AXAGON ADS-1PS, USB2.0 - serial RS232 DB9 with PREMIUMCORD Reduction serial port 9F / 9F, short. 
I managed to establish communication via this cable but I could not establish a connection to make a backup via PuTTY and I could not do it via CartridgeWriter

**Reply by mayrthom** [reply]
So you use the USB to Serial cable with a gender changer?
A gender changer doesn't work here. You have to use either a null modem cable or a null modem adapter.

**Reply by Artisan401** [reply]
Yes, I have a cable and a separate gender changer for it. So what to get? Instead of a gender changer? 
Can I get a cable with a female terminal and a USB transfer or can I get an extension cable female / female to my cable? How's the video?https://www.youtube.com/watch?v=_YLUCKgAaJk&t=107s

**Reply by mayrthom** [reply]
It has to be a **null modem cable**.

A null modem cable is a female to female cable, but internally the Rx and Tx pins are switched. So a standard female to female cable won't work and also an extension cable won't.

You just have to go to your electronic supply store of your choice and buy a RS232 null modem cable.
With this null modem cable you can connect the Printer to the USB Adapter.

I didn't make this video and it’s about an older version of this program. But the way he explained how to connect the printer is correct.

**Reply by Artisan401** [reply]
Great. So I'll just buy an RS232 null modem cable and use my already purchased RS232 to USB adapter and it should work? I would not like to buy something wrong :)

**Reply by mayrthom** [reply]
Yes this combination should work fine.

**Reply by S1L3NTANG3L** [reply]
Couldn't locally find a cable so had to build one. Ended up using this example if someone has the same issue
![698c52dd-43e3-470f-912f-0d5ef19f3ac0](https://user-images.githubusercontent.com/37051908/165111158-e6da5e9a-2b61-4e04-bf50-6b4c77df4d85.JPG)

**Comment by stefan064** [comment]
The printer also uses hardware handshaking so the null modem cable needs the RTS/CTS (and/or DTR/DSR) connections made, The printer won't send any data if only TX/RX are connected.
